--- configure.ac	2009-12-10 12:09:25.000000000 -0500
+++ configure.ac	2010-03-06 11:12:00.000000000 -0500
@@ -4249,20 +4254,14 @@
       fi ])
   AC_PATH_PROG(SDL12_CONFIG, sdl12-config, no, ${SDL_PATH})
   SDL_CONFIG="${SDL12_CONFIG}"
-  SDL_HEADER="SDL12/SDL.h"
-  SDL_IMAGE="SDL12/SDL_image.h"
   if test "${SDL_CONFIG}" = "no"
   then
     AC_PATH_PROG(SDL11_CONFIG, sdl11-config, no, ${SDL_PATH})
     SDL_CONFIG=${SDL11_CONFIG}
-    SDL_HEADER="SDL11/SDL.h"
-    SDL_IMAGE="SDL11/SDL_image.h"
   fi
   if test "${SDL_CONFIG}" = "no"
   then
     AC_PATH_PROG(SDL_CONFIG, sdl-config, no, ${SDL_PATH})
-    SDL_HEADER="SDL/SDL.h"
-    SDL_IMAGE="SDL/SDL_image.h"
   fi
   # check for cross-compiling
   SDL_PREFIX=
@@ -4281,20 +4280,16 @@
       VLC_ADD_PLUGIN([vout_sdl])
       VLC_ADD_PLUGIN([aout_sdl])
     fi
-    VLC_ADD_CFLAGS([vout_sdl aout_sdl sdl_image],[`${SDL_CONFIG} ${SDL_PREFIX} --cflags | sed 's,SDL,,'`])
+    VLC_ADD_CFLAGS([vout_sdl aout_sdl sdl_image],[`${SDL_CONFIG} ${SDL_PREFIX} --cflags`])
     VLC_ADD_LIBS([vout_sdl aout_sdl sdl_image],[`${SDL_CONFIG} ${SDL_PREFIX} --libs | sed 's,-rdynamic,,'`])
     CPPFLAGS="${CPPFLAGS_save} ${CFLAGS_vout_sdl}"
-    AC_CHECK_HEADERS(${SDL_HEADER}, AC_DEFINE_UNQUOTED(SDL_INCLUDE_FILE,
-      <${SDL_HEADER}>, Indicate whether we should use SDL/SDL.h or SDL11/SDL.h),
-      [ AC_CHECK_HEADERS(SDL.h, AC_DEFINE(SDL_INCLUDE_FILE, <SDL.h>,
-          As a last resort we also test for SDL.h presence),
-      [ AC_MSG_ERROR([The development package for SDL is not installed.
+    AC_CHECK_HEADERS(SDL.h, [], [
+      AC_MSG_ERROR([The development package for SDL is not installed.
 Please install it and try again. Alternatively you can also configure with
 --disable-sdl.])
-      ])])
+      ])
     AS_IF([ test "${enable_sdl_image}" != "no"],[
-      AC_CHECK_HEADERS(${SDL_IMAGE}, [AC_DEFINE_UNQUOTED(SDL_IMAGE_INCLUDE_FILE,
-        <${SDL_IMAGE}>, Indicate the path of SDL_image.h)
+      AC_CHECK_HEADERS(SDL_image.h, [
         VLC_ADD_PLUGIN([sdl_image])
         AC_CHECK_LIB(png, png_set_rows,
           [VLC_ADD_LIBS([sdl_image],[-lpng -lz])],[],[-lz])
@@ -4303,11 +4298,9 @@
         AC_CHECK_LIB(tiff, TIFFClientOpen,
           [VLC_ADD_LIBS([sdl_image],[-ltiff])])
         VLC_ADD_LIBS([sdl_image], [-lSDL_image])],
-        [ AC_CHECK_HEADERS(SDL_image.h, AC_DEFINE(SDL_IMAGE_INCLUDE_FILE, <SDL_image.h>,
-            As a last resort we also test for SDL_image.h presence),
         [ AC_MSG_WARN([The development package for SDL_image is not installed.
   You should install it alongside your SDL package.])
-        ])])
+        ])
     ])
     CPPFLAGS="${CPPFLAGS_save}"
     if expr 1.1.5 \> `${SDL_CONFIG} --version` >/dev/null
--- modules/codec/sdl_image.c	2009-12-07 16:12:57.000000000 -0500
+++ modules/codec/sdl_image.c	2010-03-06 11:21:18.000000000 -0500
@@ -33,7 +33,7 @@
 #include <vlc_codec.h>
 #include <vlc_vout.h>
 
-#include SDL_IMAGE_INCLUDE_FILE
+#include <SDL_image.h>
 
 /*****************************************************************************
  * decoder_sys_t : sdl decoder descriptor
--- modules/audio_output/sdl.c	2009-12-07 16:12:57.000000000 -0500
+++ modules/audio_output/sdl.c	2010-03-06 11:05:57.000000000 -0500
@@ -37,7 +37,7 @@
 #include <vlc_plugin.h>
 #include <vlc_aout.h>
 
-#include SDL_INCLUDE_FILE
+#include <SDL.h>
 
 #define FRAME_SIZE 2048
 
--- modules/video_output/sdl.c	2009-12-07 16:12:58.000000000 -0500
+++ modules/video_output/sdl.c	2010-03-06 11:05:57.000000000 -0500
@@ -45,7 +45,7 @@
 #   include <netinet/in.h>                            /* BSD: struct in_addr */
 #endif
 
-#include SDL_INCLUDE_FILE
+#include <SDL.h>
 
 /* SDL is not able to crop overlays - so use only 1 direct buffer */
 #define SDL_MAX_DIRECTBUFFERS 1
