Author: Nathan Phillip Brink <ohnobinki@ohnopublishing.net>
Date: 2010/11/27

Use libstrl for the strlcat(), strlcpy(), and strnlen() functions.

--- a/Makefile.in	Sat Nov 27 09:44:06 2010 -0500
+++ b/Makefile.in	Sat Nov 27 10:40:23 2010 -0500
@@ -34,7 +34,7 @@
 #
 
 #XCFLAGS=-O -g -export-dynamic
-IRCDLIBS=@IRCDLIBS@ @TRE_LIBS@ @CARES_LIBS@
+IRCDLIBS=@IRCDLIBS@ @TRE_LIBS@ @CARES_LIBS@ @LIBSTRL_LIBS@
 CRYPTOLIB=@CRYPTOLIB@
 OPENSSLINCLUDES=
 
--- a/configure.ac	Sat Nov 27 09:44:06 2010 -0500
+++ b/configure.ac	Sat Nov 27 10:40:23 2010 -0500
@@ -300,10 +300,13 @@
 AC_CHECK_FUNCS(vsnprintf,
 	AC_DEFINE([HAVE_VSNPRINTF], [], [Define if you have vsnprintf]))
-AC_CHECK_FUNCS(strlcpy,
-	AC_DEFINE([HAVE_STRLCPY], [], [Define if you have strlcpy. Otherwise, an internal implementation will be used!]))
-AC_CHECK_FUNCS(strlcat,
-	AC_DEFINE([HAVE_STRLCAT], [], [Define if you have strlcat]))
 AC_CHECK_FUNCS(strlncat,
 	AC_DEFINE([HAVE_STRLNCAT], [], [Define if you have strlncat]))
+
+PKG_CHECK_MODULES([LIBSTRL], [libstrl >= 0.2])
+LIBS="$LIBS $LIBSTRL_LIBS"
+CFLAGS="$CFLAGS $LIBSTRL_CFLAGS"
+AC_DEFINE([HAVE_STRLCPY], [], [Define if you have strlcpy(). Otherwise, an internal implementation will be used!])
+AC_DEFINE([HAVE_STRLCAT], [], [Define if you have strlcat().])
+
 AC_CHECK_FUNCS(inet_pton,
 	AC_DEFINE([HAVE_INET_PTON], [], [Define if you have inet_pton]))
