From http://www.selenic.com/mercurial/bts/file636/modwsgi.diff, http://www.selenic.com/mercurial/bts/issue1095, by chmullig 2008/06/08
diff -r d90d83ebea9e mercurial/ui.py
--- a/mercurial/ui.py   Tue Jun 03 12:10:14 2008 +0200
+++ b/mercurial/ui.py   Sat Jun 07 22:58:36 2008 -0400
@@ -385,12 +385,14 @@
 
     def write_err(self, *args):
         try:
-            if not sys.stdout.closed: sys.stdout.flush()
+            try: sys.stdout.flush()
+            except: pass
             for a in args:
                 sys.stderr.write(str(a))
             # stderr may be buffered under win32 when redirected to files,
             # including stdout.
-            if not sys.stderr.closed: sys.stderr.flush()
+            try: sys.stderr.flush()
+            except: pass
         except IOError, inst:
             if inst.errno != errno.EPIPE:
                 raise
