diff -u /mnt/satahd/portagetmp/portage/media-libs/libdvdnav-4.1.2/work/libdvdnav-4.1.2/src/highlight.c /tmp/buffer-content-24939WGo
*** /libdvdnav-4.1.2/src/highlight.c	2008-06-28 12:28:46.000000000 -0400
--- /tmp/buffer-content-24939WGo	2008-06-28 12:29:19.000000000 -0400
***************
*** 253,258 ****
--- 253,260 ----
  
  dvdnav_status_t dvdnav_upper_button_select(dvdnav_t *this, pci_t *pci) {
    btni_t *button_ptr;
+ 
+   printf(MSG_OUT, "upper button\n");
    
    if(!(button_ptr = get_current_button(this, pci)))
      return DVDNAV_STATUS_ERR;
***************
*** 263,268 ****
--- 265,272 ----
  
  dvdnav_status_t dvdnav_lower_button_select(dvdnav_t *this, pci_t *pci) {
    btni_t *button_ptr;
+ 
+   printf(MSG_OUT, "lower button\n");
    
    if(!(button_ptr = get_current_button(this, pci)))
      return DVDNAV_STATUS_ERR;
***************
*** 273,278 ****
--- 277,284 ----
  
  dvdnav_status_t dvdnav_right_button_select(dvdnav_t *this, pci_t *pci) {
    btni_t *button_ptr;
+ 
+   printf(MSG_OUT, "right button\n");
    
    if(!(button_ptr = get_current_button(this, pci)))
      return DVDNAV_STATUS_ERR;
***************
*** 283,288 ****
--- 289,296 ----
  
  dvdnav_status_t dvdnav_left_button_select(dvdnav_t *this, pci_t *pci) {
    btni_t *button_ptr;
+ 
+   printf(MSG_OUT, "left button\n");
    
    if(!(button_ptr = get_current_button(this, pci)))
      return DVDNAV_STATUS_ERR;

Diff finished.  Sat Jun 28 12:29:20 2008
