diff -u /mnt/satahd/portagetmp/portage/media-libs/libdvdnav-4.1.2/work/libdvdnav-4.1.2/src/Makefile.am /tmp/buffer-content-26498N6Q
--- /libdvdnav-4.1.2/src/Makefile.am	2008-06-25 16:12:16.000000000 -0400
+++ /libdvdnav-4.1.2/src/Makefile.am	2008-06-25 16:12:28.000000000 -0400
@@ -1,10 +1,10 @@
 include $(top_srcdir)/misc/Makefile.common
 
-SUBDIRS = dvdread vm
+SUBDIRS = vm
 
 includedir = ${prefix}/include/dvdnav
 
 AM_CPPFLAGS = -DDVDNAV_COMPILE $(THREAD_CFLAGS) \
-	-I$(top_srcdir)/src/dvdread -I$(top_srcdir)/src/vm
+	-I${prefix}/include/dvdread -I$(top_srcdir)/src/vm
 
 EXTRA_DIST = README.MAP FELLOWSHIP.map
@@ -17,7 +17,6 @@
 	dvdnav_internal.h read_cache.h remap.h
 
 libdvdnav_la_LIBADD = $(THREAD_LIBS) \
-	$(top_builddir)/src/dvdread/libdvdread.la \
 	$(top_builddir)/src/vm/libdvdvm.la
 
 libdvdnav_la_LDFLAGS = \

diff -u /mnt/satahd/portagetmp/portage/media-libs/libdvdnav-4.1.2/work/libdvdnav-4.1.2/configure.ac /tmp/buffer-content-26498nOd
--- /libdvdnav-4.1.2/configure.ac	2007-05-01 06:11:52.000000000 -0400
+++ /libdvdnav-4.1.2/configure.ac	2008-06-25 16:41:21.000000000 -0400
@@ -224,7 +224,6 @@
 AC_OUTPUT([
 Makefile 
 src/Makefile
-src/dvdread/Makefile
 src/vm/Makefile
 misc/Makefile
 misc/dvdnav-config
diff -u /libdvdnav-4.1.2/src/vm/Makefile.am /tmp/buffer-content-24939W_z
--- /libdvdnav-4.1.2/src/vm/Makefile.am	2008-06-27 10:42:33.000000000 -0400
+++ /libdvdnav-4.1.2/src/vm/Makefile.am	2008-06-27 10:43:03.000000000 -0400
@@ -3,6 +3,6 @@
 includedir = ${prefix}/include/dvdnav
 
 AM_CPPFLAGS = -DDVDNAV_COMPILE $(THREAD_CFLAGS) \
-	-I$(top_srcdir)/src -I$(top_srcdir)/src/dvdread
+	-I$(top_srcdir)/src -I${prefix}/include/dvdread
 
 noinst_LTLIBRARIES = libdvdvm.la


diff -u /mnt/satahd/portagetmp/portage/media-libs/libdvdnav-4.1.2/work/libdvdnav-4.1.2/examples/Makefile.am /tmp/buffer-content-24939IJD
--- /libdvdnav-4.1.2/examples/Makefile.am	2004-01-11 16:43:13.000000000 -0500
+++ /tmp/buffer-content-24939IJD	2008-06-27 11:33:00.000000000 -0400
@@ -1,6 +1,6 @@
 include $(top_srcdir)/misc/Makefile.common
 
-AM_CPPFLAGS = -DDVDNAV_COMPILE -I$(top_srcdir)/src -I$(top_srcdir)/src/dvdread
+AM_CPPFLAGS = -DDVDNAV_COMPILE -I$(top_srcdir)/src -I${prefix}/include/dvdread
 
 DVDNAV_LIB = $(top_builddir)/src/libdvdnav.la
 
