diff -u /var/tmp/portage/media-gfx/povray-3.7.0_beta29-r1/work/povray-3.7.0.beta.29/vfe/unix/unixconsole.cpp /tmp/buffer-content-10159TsY
--- povray-3.7.0.beta.29/vfe/unix/unixconsole.cpp	2008-10-07 19:06:27.000000000 -0400
+++ /tmp/buffer-content-10159TsY	2009-02-23 22:36:04.000000000 -0500
@@ -254,6 +254,7 @@
 
 void PrintBeta(int argc, char **argv)
 {
+#if 0
 	const int sph = 3600, spd = 86400, spw = 604800;
 	time_t current_time = time(NULL);
 
@@ -296,6 +297,7 @@
 		"%s: this pre-release version of POV-Ray for Unix expires in %d day(s) and %d hour(s)\n",
 		PACKAGE, current_time/spd, (current_time % spd)/sph
 	);
+#endif
 }
 
 void PrintVersion(void)

Diff finished.  Mon Feb 23 22:36:05 2009
