diff -ru /var/tmp/portage/dev-libs/confuse-2.6-r3/work/confuse-2.6/Makefile.am ./Makefile.am
--- ./Makefile.am	2007-10-13 16:39:59.000000000 -0400
+++ ./Makefile.am	2009-02-28 22:27:06.000000000 -0500
@@ -3,7 +3,11 @@
 AUTOMAKE_OPTIONS = foreign
 CLEANFILES=*~ '\#*\#'
 EXTRA_DIST=libconfuse.spec.in libconfuse.spec
-SUBDIRS = m4 po src examples tests doc
+
+if ENABLE_EXAMPLES
+  EXAMPLES = examples
+endif
+SUBDIRS = m4 po src $(EXAMPLES) tests doc
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libconfuse.pc
diff -ru /var/tmp/portage/dev-libs/confuse-2.6-r3/work/confuse-2.6/configure.ac ./configure.ac
--- ./configure.ac	2009-02-28 20:54:04.000000000 -0500
+++ ./configure.ac	2009-02-28 22:26:38.000000000 -0500
@@ -24,6 +24,13 @@
 AC_DISABLE_SHARED
 AC_PROG_LIBTOOL
 
+#optional building of examples:
+AC_ARG_ENABLE(examples, 
+	AC_HELP_STRING([--disable-examples], [don't build examples in examples]),
+	[], [enable_examples=yes])
+
+AM_CONDITIONAL([ENABLE_EXAMPLES], [test "$enable_examples" = yes])
+echo enable_examples=$enable_examples
 AM_CONDITIONAL([cc_is_gcc], test "$GCC" = "yes")
 
 AM_GNU_GETTEXT([external])
