diff -c /mnt/bacup/portagetmp/portage/media-libs/sdl-image-1.2.6-r1/work/SDL_image-1.2.6/Makefile.am /var/tmp/portage/media-libs/sdl-image-1.2.6-r1/work/SDL_image-1.2.6/Makefile.am
--- /Makefile.am	2007-07-21 01:26:34.000000000 -0400
+++ /var/tmp/portage/media-libs/sdl-image-1.2.6-r1/work/SDL_image-1.2.6/Makefile.am	2008-09-17 21:29:38.000000000 -0400
@@ -39,14 +39,14 @@
 	-no-undefined		\
 	-release $(LT_RELEASE)	\
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) -Wl,version.o
-libSDL_image_la_LIBADD = $(IMG_LIBS)
+libSDL_image_la_LIBADD = $(LIBS)
 libSDL_image_la_DEPENDENCIES = version.o
 else
 libSDL_image_la_LDFLAGS = 	\
 	-no-undefined		\
 	-release $(LT_RELEASE)	\
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
-libSDL_image_la_LIBADD = $(IMG_LIBS)
+libSDL_image_la_LIBADD = $(LIBS)
 endif
 
 %.o : %.rc
@@ -54,6 +54,6 @@
 
 noinst_PROGRAMS = showimage
 
-showimage_LDADD = libSDL_image.la
+showimage_LDADD = libSDL_image.la $(LIBS) 
 
 # Rule to build tar-gzipped distribution package

Diff finished.  Wed Sep 17 21:32:09 2008
