--- /madwifi-0.9.4/ath/if_ath_pci.c	2008-01-30 15:56:11.000000000 -0500
+++ /madwifi-0.9.4/ath/if_ath_pci.c	2008-06-23 20:44:49.000000000 -0400
@@ -169,6 +169,6 @@
 
 	phymem = pci_resource_start(pdev, 0);
 	if (!request_mem_region(phymem, pci_resource_len(pdev, 0), "ath")) {
-		printk(KERN_ERR "ath_pci: cannot reserve PCI memory region\n");
+		  printk(KERN_ERR "ath_pci: cannot reserve PCI memory region at %lx-%llx, length%llx\n", phymem, phymem + pci_resource_len(pdev, 0), pci_resource_len(pdev, 0));
 		goto bad;
 	}
