diff -c /home/ohnobinki/ivd/phidgetprogs/phidgetlinux/Phidgetlinux/phidget21/Makefile /tmp/buffer-content-3770fbg
--- Makefile	Mon Sep 22 14:48:19 2008
+++ -content-3770fbg	Tue Sep 30 09:27:29 2008
@@ -150,7 +150,7 @@
 	cd Java ; jar cf ../$@ com/phidgets/event/*.class com/phidgets/*.class
 
 install: all
-	$(INSTALL_LIB) libphidget21.so $(INSTALLPREFIX)/$(PREFIX)/lib
+	$(INSTALL_LIB) libphidget21.so $(INSTALLPREFIX)/$(PREFIX)/$(LIBDIR)
 	$(INSTALL_FILE) phidget21.h $(INSTALLPREFIX)/$(PREFIX)/include
 
 uninstall:

Diff finished.  Tue Sep 30 09:27:29 2008
