--- glame-2.0.1/src/include/glame_guile_compat.h	Sat Oct 23 09:09:27 2004
+++ glame-2.0.1/src/include/glame_guile_compat.h	Tue Mar 10 18:49:36 2009
@@ -47,7 +47,7 @@
 #define glame_scm2double(obj) scm_num2dbl (obj, "somewhere")
 #define scm_str2string(str) gh_str02scm (str)
 
-#if (SCM_MINOR_VERSION == 7)
+#if (SCM_MINOR_VERSION > 6)
 #define GLAME_NEWCELL(z) z = scm_cell (SCM_UNPACK(z), 0)
 #endif
 

