diff -r 6d373ecb8815 Makefile.in
--- a/Makefile.in	Mon Feb 15 16:46:04 2010 -0600
+++ b/Makefile.in	Mon Feb 15 20:57:33 2010 -0500
@@ -35,11 +35,11 @@
 SUBDIRS=@LIBMOWGLI@ modules src po
 CLEANDIRS = ${SUBDIRS}
 
-.PHONY: dist
+.PHONY: dist depend
 
-all: depend build
+all: build
 
-build: include/hooktypes.h
+build: depend
 	-@if [ ! -f include/sysconf.h ] ; then \
 		echo "Hmm...doesn't look like you've run configure..."; \
 		echo "Doing so now."; \
